tfe/data_source_agent_pool.go

```diff
@@ -35,10 +46,19 @@ func dataSourceTFEAgentPoolRead(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
-	id, err := fetchAgentPoolID(organization, name, config.Client)
+	pool, err := fetchAgentPool(organization, name, config.Client)
 	if err != nil {
 		return err
 	}
-	d.SetId(id)
+
+	d.SetId(pool.ID)
+	d.Set("organization_scoped", pool.OrganizationScoped)
+
+	var wids []interface{}
+	for _, workspace := range pool.AllowedWorkspaces {
+		wids = append(wids, workspace.ID)
```

Comment:
In that case, we might want to rename this variable `allowedWorkspaceIDs` since there is a different `workspaces` relationship on the Agent Pool object. It's not consumable currently but we may add that support in the future.